tests/run_tox.py

```diff
@@ -0,0 +1,167 @@
+    # try to remove them for real
+    containers_to_remove = [item.strip() for item in
+                            sh('docker ps -a -q', capture=True)[0].split()]
+    for container in containers_to_remove:
+        sh('docker rm -f %s' % container, ignore_errors=True)
+
+    for subdir in ['daemon', 'demo', 'base']:
+        os.mkdir(os.path.join(env['WORKSPACE'], subdir))
+        # starting with kraken, the base image does not exist
+        cmd = 'cp -Lrv ceph-releases/{0}/{1}/{2}/* {2}'.format(
+            env['CEPH_STABLE_RELEASE'], env['IMAGE_DISTRO'], subdir)
+        sh(cmd, ignore_errors=(subdir != 'daemon'))
+
+    sh('bash "%s"/travis-builds/build_imgs.sh' % env['WORKSPACE'])
+
+    # start a local docker registry
+    sh('docker run -d -p 5000:5000 --restart=always --name registry registry:2')  # noqa
+    # add the image we just built to the registry
+    docker_img = 'localhost:5000/ceph/daemon:%s-latest' % \
+        env['CEPH_STABLE_RELEASE']
+
+    sh('docker tag ceph/daemon %s' % docker_img)
+    # this avoids a race condition between the tagging and the push
+    # which causes this to sometimes fail when run by jenkins
+    time.sleep(1)
+    sh('docker --debug push %s' % docker_img)
+
+
+def test(env):
+    # TODO: get the output image from build_imgs.sh to pass onto ceph-ansible
+
+    # run vagrant and ceph-ansible tests
+    cwd = env['CEPH_ANSIBLE_SCENARIO_PATH']
+    sh('vagrant up --no-provision --provider=%s' % env['VAGRANT_PROVIDER'],
+       cwd=cwd)
+
+    sh('bash %s/ceph-ansible/tests/scripts/generate_ssh_config.sh %s' %
+       (env['TOXINIDIR'], env['CEPH_ANSIBLE_SCENARIO_PATH']), cwd=cwd)
+
+    env['ANSIBLE_SSH_ARGS'] = "-F %s/vagrant_ssh_config" % env['CEPH_ANSIBLE_SCENARIO_PATH']
+
+    # runs a playbook to configure nodes for testing
+    hosts_file = "%s/hosts" % env['CEPH_ANSIBLE_SCENARIO_PATH']
+    ansible_templ = "ansible-playbook -vv -i {hosts} {therest}"
+    ansible_playbook_args = [
+        "{TOXINIDIR}/tests/setup.yml",
+        "{TOXINIDIR}/ceph-ansible/site-docker.yml.sample --extra-vars='ceph_stable_release={CEPH_STABLE_RELEASE} ceph_docker_image_tag={CEPH_STABLE_RELEASE}-latest ceph_docker_registry={REGISTRY_ADDRESS} fetch_directory={CEPH_ANSIBLE_SCENARIO_PATH}/fetch'",
+        "{TOXINIDIR}/ceph-ansible/tests/functional/setup.yml",
+    ]
+    for pb_args in ansible_playbook_args:
+        pb_args = pb_args.format(**env)
+        sh(ansible_templ.format(hosts=hosts_file, therest=pb_args), cwd=cwd)
+
+    sh("testinfra -n 4 --sudo -v --connection=ansible --ansible-inventory={CEPH_ANSIBLE_SCENARIO_PATH}/hosts {TOXINIDIR}/ceph-ansible/tests/functional/tests".format(**env),
+       cwd=cwd)
+
+
+def teardown(env):
+    sh("{TOXINIDIR}/tests/teardown.sh".format(**env),
+        cwd=env['CEPH_ANSIBLE_SCENARIO_PATH'])
+
+
+def main():
+    env = os.environ
+    setup(env)
+    try:
+        test(env)
+    except Exception:
```

Comment:
Oh, yeah, that is certainly vestigial. As I mention [here](https://github.com/ceph/ceph-docker/pull/659#issuecomment-305571306) this PR isn't mature this very second. I left that stanza there because I was considering adding some error handling and/or reporting.